CMakeLists.txt

```diff
@@ -782,7 +787,7 @@ target_include_directories(
 set_property(TARGET irods_common PROPERTY CXX_STANDARD ${IRODS_CXX_STANDARD})
 set_property(TARGET irods_common PROPERTY VERSION ${IRODS_VERSION})
 set_property(TARGET irods_common PROPERTY SOVERSION ${IRODS_VERSION})
-target_compile_definitions(irods_common PRIVATE ${IRODS_COMPILE_DEFINITIONS})
+target_compile_definitions(irods_common PRIVATE ${IRODS_COMPILE_DEFINITIONS} RODS_SERVER SPDLOG_ENABLE_SYSLOG)
```

Comment:
do we want RODS_SERVER here?

Reply:
iRODS cannot be compiled without it mainly because of libRodsAPIs.  The RODS_SERVER macro is used to enable code that forwards messages from rodsLog to the new logging API.  It can be handled in other ways, it's just a matter of which way we prefer.

Is libRodsAPIs still needed?

If libRodsAPIs is no longer supported and we remove it, then I think the RODS_SERVER macro can be removed.

Reply:
Code Update:
- Resolved compilation issues with libRodsAPIs.
- Removed dependency on RODS_SERVER macro.
- Moved SPDLOG_ENABLE_SYSLOG macro into header file and introduced  IRODS_ENABLE_SYSLOG macro for control.